apps/dashboard/pkg/migration/conversion/v2beta1_to_v2alpha1.go

```diff
@@ -0,0 +1,997 @@
+				AllValue:         in.DatasourceVariableKind.Spec.AllValue,
+				Label:            in.DatasourceVariableKind.Spec.Label,
+				Hide:             dashv2alpha1.DashboardVariableHide(in.DatasourceVariableKind.Spec.Hide),
+				SkipUrlSync:      in.DatasourceVariableKind.Spec.SkipUrlSync,
+				Description:      in.DatasourceVariableKind.Spec.Description,
+				AllowCustomValue: in.DatasourceVariableKind.Spec.AllowCustomValue,
+			},
+		}
+	}
+
+	if in.IntervalVariableKind != nil {
+		out.IntervalVariableKind = &dashv2alpha1.DashboardIntervalVariableKind{
+			Kind: in.IntervalVariableKind.Kind,
+			Spec: dashv2alpha1.DashboardIntervalVariableSpec{
+				Name:        in.IntervalVariableKind.Spec.Name,
+				Query:       in.IntervalVariableKind.Spec.Query,
+				Current:     convertVariableOption_V2beta1_to_V2alpha1(in.IntervalVariableKind.Spec.Current),
+				Options:     convertVariableOptions_V2beta1_to_V2alpha1(in.IntervalVariableKind.Spec.Options),
+				Auto:        in.IntervalVariableKind.Spec.Auto,
+				AutoMin:     in.IntervalVariableKind.Spec.AutoMin,
+				AutoCount:   in.IntervalVariableKind.Spec.AutoCount,
+				Refresh:     dashv2alpha1.DashboardVariableRefresh("load"), // Interval variables always use "load" in v2alpha1
+				Label:       in.IntervalVariableKind.Spec.Label,
+				Hide:        dashv2alpha1.DashboardVariableHide(in.IntervalVariableKind.Spec.Hide),
+				SkipUrlSync: in.IntervalVariableKind.Spec.SkipUrlSync,
+				Description: in.IntervalVariableKind.Spec.Description,
+			},
+		}
+	}
+
+	if in.CustomVariableKind != nil {
+		out.CustomVariableKind = &dashv2alpha1.DashboardCustomVariableKind{
+			Kind: in.CustomVariableKind.Kind,
+			Spec: dashv2alpha1.DashboardCustomVariableSpec{
+				Name:             in.CustomVariableKind.Spec.Name,
+				Query:            in.CustomVariableKind.Spec.Query,
+				Current:          convertVariableOption_V2beta1_to_V2alpha1(in.CustomVariableKind.Spec.Current),
+				Options:          convertVariableOptions_V2beta1_to_V2alpha1(in.CustomVariableKind.Spec.Options),
+				Multi:            in.CustomVariableKind.Spec.Multi,
+				IncludeAll:       in.CustomVariableKind.Spec.IncludeAll,
+				AllValue:         in.CustomVariableKind.Spec.AllValue,
+				Label:            in.CustomVariableKind.Spec.Label,
+				Hide:             dashv2alpha1.DashboardVariableHide(in.CustomVariableKind.Spec.Hide),
+				SkipUrlSync:      in.CustomVariableKind.Spec.SkipUrlSync,
+				Description:      in.CustomVariableKind.Spec.Description,
+				AllowCustomValue: in.CustomVariableKind.Spec.AllowCustomValue,
+			},
+		}
+	}
+
+	if in.GroupByVariableKind != nil {
+		// Extract datasource from top-level fields
+		var dsType *string
+		var dsUID *string
+		if in.GroupByVariableKind.Datasource != nil && in.GroupByVariableKind.Datasource.Name != nil {
+			dsUID = in.GroupByVariableKind.Datasource.Name
+		}
+		if in.GroupByVariableKind.Group != "" {
+			dsType = &in.GroupByVariableKind.Group
+		}
+
```

Comment:
should we add a check here either `dsUID` or `dsType` are set? otherwise the datasource ref can be empty - is that okay?